docs/architecture/ErrorReporting.md

```diff
@@ -0,0 +1,140 @@
+# Handling and reporting errors
+
+    WARNING: This document describes future plans for error handling.
+    The UI library does not (yet) implement error handling as described here.
+
+This document describes how errors are handled within the ACS UI library and how they are reported to the surrounding application.
+
+## Error sources
+
+There are two categories of errors, and they are handled differently.
+
+1. ACS backend errors: Errors returned by the REST calls to ACS backend services.
```

Comment:
I think this doc needs to be agnostic of ACS as the adapter interface isn't coupled to ACS

Reply:
Thanks for bringing up that adapter isn't tied to ACS, and that it holds all state for the UI. This has real technical implications (e.g., the `ErrorBar` component must allow for a way to extend the error -> string mapping to include errors not from stateful).

This document must talk about ACS, because it applies to error reporting throughout the library, including from the stateful layer / components-plus-bindings use-case.

But I've reworked the composite section to include the case when the adapter isn't backed by ACS.